js/lib/SparseMerkleTree.js

```diff
@@ -0,0 +1,147 @@
+const { sha3, sha256, bufferToHex } = require('ethereumjs-util');
+const BigNumber = require('bignumber.js');
+
+const hash = sha3;
+
+class SparseMerkleTree {
```

Comment:
Long-term we probably need better caching strategies.

Reply:
we should defiantly implement a way to append a leaf W/O rebuilding unaffected parts